File: lib/src/models/staff_layout.dart
import 'package:meta/meta.dart';

@immutable
class StaffLayout {
  final int
      staffNumber; // staff number this layout applies to (default 1 if not present in XML attribute)
  final double? staffDistance; // distance from previous staff

  const StaffLayout({
    required this.staffNumber,
    this.staffDistance,
  });

  @override
  bool operator ==(Object other) =>
      identical(this, other) ||
      other is StaffLayout &&
          runtimeType == other.runtimeType &&
          staffNumber == other.staffNumber &&
          staffDistance == other.staffDistance;

  @override
  int get hashCode => staffNumber.hashCode ^ staffDistance.hashCode;

  @override
  String toString() {
    return 'StaffLayout{staffNumber: $staffNumber, staffDistance: $staffDistance}';
  }
}

File: MusicXMLParser/Models/StaffLayout.cs
using System;

namespace MusicXMLParser.Models
{
    public class StaffLayout
    {
        /// <summary>
        /// Staff number this layout applies to (default 1 if not present in XML attribute).
        /// </summary>
        public int StaffNumber { get; }

        /// <summary>
        /// Distance from the previous staff.
        /// </summary>
        public double? StaffDistance { get; }

        public StaffLayout(int staffNumber, double? staffDistance = null)
        {
            StaffNumber = staffNumber;
            StaffDistance = staffDistance;
        }

        public override bool Equals(object? obj)
        {
            if (obj is StaffLayout other)
            {
                return StaffNumber == other.StaffNumber &&
                       StaffDistance == other.StaffDistance;
            }
            return false;
        }

        public override int GetHashCode()
        {
            return HashCode.Combine(StaffNumber, StaffDistance);
        }

        public override string ToString()
        {
            return $"StaffLayout{{StaffNumber: {StaffNumber}, StaffDistance: {StaffDistance?.ToString() ?? "null"}}}";
        }
    }
}
